modules/logging/logging.go

```diff
@@ -35,9 +37,21 @@ func SharedInstance() *LogStruct {
 	return sharedInstance
 }
 
-func (u *LogStruct) MethodInfo(model string, method string) *logrus.Entry {
-	return u.log.WithFields(logrus.Fields{
-		"model":  model,
-		"method": method,
-	})
+func (u *LogStruct) MethodInfo(model string, method string, stack ...bool) *logrus.Entry {
+	if len(stack) > 0 && stack[0] {
+		buf := make([]byte, 1<<16)
+		runtime.Stack(buf, false)
+		return u.log.WithFields(logrus.Fields{
+			"time":       time.Now(),
+			"model":      model,
+			"method":     method,
+			"stacktrace": string(buf),
+		})
+	} else {
```

Comment:
__[golint]__
* [golint] if block ends with a return statement, so drop this else and outdent its block